core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/InternalTerms.java

```diff
@@ -246,7 +246,16 @@ public InternalAggregation doReduce(List<InternalAggregation> aggregations, Redu
             }
             setDocCountError(thisAggDocCountError);
             for (B bucket : terms.getBucketsInternal()) {
-                bucket.docCountError = thisAggDocCountError;
+                if (bucket.docCountError > 0) {
+                    // If there is already a doc count error for this bucket
+                    // subtract it from this aggs doc count error to make the
+                    // new value for the bucket. This then means that when the
+                    // final error for the bucket is calculated below we account
+                    // for the existing error calculated in a previous reduce.
+                    bucket.docCountError -= thisAggDocCountError;
+                } else {
+                    bucket.docCountError = thisAggDocCountError;
+                }
```

Comment:
this should be `bucket.docCountError -= thisAggDocCountError;` in both branches I think?